### PR TITLE
Add root info

### PR DESCRIPTION
**Pull request by Jesus**

### CODE REVIEW

**Comment by Jesus** [comment]
@dwhenry I don't have a Dropbox business account, so I could only test to a certain point. It'd be awesome if to gather feedback on this branch from people who actually use a business account.

I'll leave this open for a few days in case anyone has suggestions and eventually I'll merge it.

**Comment by kmuto** [comment]
Hi, I tried this branch with my Dropbox business account.
Sadly it failed.

sample code
```
#!/usr/bin/ruby
$LOAD_PATH.unshift("./lib") # for use local git version 
require 'dropbox_api'
client = DropboxApi::Client.new("...")
client.namespace_id = client.get_current_account.root_info.root_namespace_id

folder = client.list_folder('')
```

result (ruby 2.5.5p157)
```
Traceback (most recent call last):
	4: from dropboxapi.rb:8:in `<main>'
	3: from dropbox_api/lib/dropbox_api/client.rb:23:in `block in add_endpoint'
	2: from dropbox_api/lib/dropbox_api/endpoints/files/list_folder.rb:44:in `block in <class:ListFolder>'
	1: from dropbox_api/lib/dropbox_api/endpoints/base.rb:12:in `perform_request'
dropbox_api/lib/dropbox_api/endpoints/base.rb:37:in `process_response': HTTP 400: Error in call to API function "files/list_folder": Dropbox-API-Path-Root: unknown tag '#<DropboxApi::Metadata::NamespaceId:0x000056279ea03d60>' (DropboxApi::Errors::HttpError)
```

 I'll try to dig more.

File: lib/dropbox_api/middleware/path_root.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      else
+        return options[:namespace_id]
+      end
+    end
+
+    def namespace_id_header_value
+      JSON.dump(
+        DropboxApi::Metadata::NamespaceId.new({
+          'namespace_id' => namespace_id
+        })
```
Comment (kmuto) [inline lib/dropbox_api/middleware/path_root.rb:23]: by changing here to `}).to_hash`, handling team works.

**Comment by Jesus** [comment]
Thanks for the feedback @kmuto & @Nerian

**Pull request merged**